packages/bot-web-ui/src/components/dashboard/quick-strategy/quick-strategy-components/data/data-uniq-input-obj.ts

```diff
@@ -0,0 +1,53 @@
+import QuickStrategyStore from 'Stores/quick-strategy-store';
+import common_inputs_properties, { TCommonInputsProperties } from './common-input-properties';
+import { TInputUniqFields } from '../../quick-strategy.types';
+
+const quick_strategy_store = new QuickStrategyStore();
+
+export type TDataUniqInput = {
+    index: number;
+    id: TFieldNameUniqInput;
+    field_name: TFieldNameUniqInput;
+    input_value: TInputUniqFields;
+    label: TFieldNameUniqLabel;
+    placeholder: string;
+    trailing_icon_message: string;
+} & Readonly<TCommonInputsProperties>;
+
+type TFieldNameUniqInput = 'martingale-size' | 'alembert-unit' | 'oscar-unit';
+type TFieldNameUniqLabel = 'Size' | '' | 'Units';
+
+const data_uniq_input_obj: ReadonlyArray<TDataUniqInput> = [
+    {
+        index: 0,
```

Comment:
@prince-deriv index equals to the index of each object in this array, so why do we explicitly add this property? I bet it's extra and has to be removed. A regular object index can be used instead. If the intention is to add a unique identifier, then we can use the id that we have in each of these objects 🙂

Reply:
@maryia-binary we use the `index` to define the unique field by way comparing `selected_type_strategy's` index from the **config** and `index` of our element `.filter(elem => elem.index === selected_type_strategy.index)[0]`. Of course, we can rename it to `id` if it has any difference.

Reply:
@maryiafrantsava-binary these object have unique ids like `id: 'martingale-size'` 🙂
This work the same way, right?:
`data_uniq_input_obj.filter((elem, index) => index === selected_type_strategy.index)[0]`

Reply:
@maryia-binary thanks for the good suggestion 👍

Reply:
done ✅